### PR TITLE
feat: docusaurus configuration

### LINKED ISSUE #13: chore: initial configuration (for draft purposes, not for production)

### PR DESCRIPTION
**Pull request by mfw78**
# Description
This PR provides initial configuration for docusaurus. This is a necessary step to moving docusaurus into production.

# Changes

- [x] Simple configuration for docusaurus to use in github pages.
- [x] Moved legacy documentation to correct source location.

**NOTE**: Some docusaurus images are included so that the pages build before the templates are adjusted.

## Related Issues

Closes cowprotocol/docs#13

### CODE REVIEW

**Comment by mfw78** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by mfw78** [comment]
recheck

**Review by shoom3301 — APPROVED** [review]

File: docusaurus.config.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+          // Remove this to remove the "edit this page" links.
+          editUrl:
+            'https://github.com/facebook/docusaurus/tree/main/packages/create-docusaurus/templates/shared/',
+        },
+        blog: {
+          showReadingTime: true,
+          // Please change this to your repo.
+          // Remove this to remove the "edit this page" links.
+          editUrl:
+            'https://github.com/facebook/docusaurus/tree/main/packages/create-docusaurus/templates/shared/',
```
Comment (alfetopito) [inline docusaurus.config.js]: Should the urls point to this repo instead?

File: docusaurus.config.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
+// @ts-check
+// Note: type annotations allow type checking and IDEs autocompletion
+
+const lightCodeTheme = require('prism-react-renderer/themes/github');
+const darkCodeTheme = require('prism-react-renderer/themes/dracula');
+
+/** @type {import('@docusaurus/types').Config} */
+const config = {
+  title: 'CoW Protocol Documentation',
+  tagline: 'Tagline here',
```
Comment (alfetopito) [inline docusaurus.config.js:10]: Will this be updated later?

File: docusaurus.config.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+  themeConfig:
+    /** @type {import('@docusaurus/preset-classic').ThemeConfig} */
+    ({
+      // Replace with your project's social card
+      image: 'img/docusaurus-social-card.jpg',
+      navbar: {
+        title: 'CoW Protocol Documentation',
+        logo: {
+          alt: 'My Site Logo',
```
Comment (alfetopito) [inline docusaurus.config.js:62]: And here

File: docusaurus.config.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        items: [
+          {
+            type: 'docSidebar',
+            sidebarId: 'tutorialSidebar',
+            position: 'left',
+            label: 'Tutorial',
+          },
+          {to: '/blog', label: 'Blog', position: 'left'},
+          {
+            href: 'https://github.com/facebook/docusaurus',
```
Comment (alfetopito) [inline docusaurus.config.js]: And here

File: docusaurus.config.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+                label: 'Stack Overflow',
+                href: 'https://stackoverflow.com/questions/tagged/docusaurus',
+              },
+              {
+                label: 'Discord',
+                href: 'https://discordapp.com/invite/docusaurus',
+              },
+              {
+                label: 'Twitter',
+                href: 'https://twitter.com/docusaurus',
```
Comment (alfetopito) [inline docusaurus.config.js:95]: And here :)

File: docusaurus.config.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+                to: '/blog',
+              },
+              {
+                label: 'GitHub',
+                href: 'https://github.com/facebook/docusaurus',
+              },
+            ],
+          },
+        ],
+        copyright: `Copyright © ${new Date().getFullYear()} My Project, Inc. Built with Docusaurus.`,
```
Comment (alfetopito) [inline docusaurus.config.js:109]: And so on

File: package.json [hunk 7]
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "cow-docs",
+  "version": "0.0.0",
+  "private": true,
+  "scripts": {
+    "docusaurus": "docusaurus",
```
Comment (alfetopito) [inline package.json:6]: Is this command needed?

File: src/components/HomepageFeatures/index.js
Comment (alfetopito) [inline src/components/HomepageFeatures/index.js:1]: And this one too.

I guess the content will be handled later

File: static/img/docusaurus-social-card.jpg
Comment (alfetopito) [inline static/img/docusaurus-social-card.jpg:1]: Maybe this shouldn't be uploaded at all?

**Review by alfetopito — APPROVED** [review]
Couple of nitpics about the content.
I know, I know, will be handled later.
I just like to mention anyway

File: docusaurus.config.js (hunk 1 above)
Comment (mfw78) [reply docusaurus.config.js]: Tracked in #18

File: docusaurus.config.js (hunk 2 above)
Comment (mfw78) [reply docusaurus.config.js:10]: Tracked in #18

File: docusaurus.config.js (hunk 3 above)
Comment (mfw78) [reply docusaurus.config.js:62]: Tracked in #18

File: docusaurus.config.js (hunk 4 above)
Comment (mfw78) [reply docusaurus.config.js]: Tracked in #18

File: docusaurus.config.js (hunk 5 above)
Comment (mfw78) [reply docusaurus.config.js:95]: Tracked in #18

File: docusaurus.config.js (hunk 6 above)
Comment (mfw78) [reply docusaurus.config.js:109]: Tracked in #18

File: package.json (hunk 7 above)
Comment (mfw78) [reply package.json:6]: Tracked in #18

File: src/components/HomepageFeatures/index.js
Comment (mfw78) [reply src/components/HomepageFeatures/index.js:1]: Tracked in #17 (I wasn't just pasting the same thing :sweat_smile:)

**Review by anxolin — COMMENTED** [review]
👍 I hope you know what u are doing, since I'm not reading the 9K changes 😂

**Comment by anxolin** [comment]
@mfw78 i tried to run this locally (in develop branch) and I got this

image


Also, should the run instructions be in the README? 

I tried this too with the same result
image

**Pull request merged**